src/move_basic.cpp

```diff
@@ -190,6 +202,51 @@ bool MoveBasic::getTransform(const std::string& from, const std::string& to,
 }
 
 
+// Called when a laser scan is received assumes laser scanner is
+// mounted at around base_link. Sets obstacleDetected
+
+void MoveBasic::scanCallback(const sensor_msgs::LaserScan::ConstPtr &msg)
+{
+    double angle = msg->angle_min;
+    double increment = msg->angle_increment;
+    double width_2 = robotWidth / 2.0;
+    double rangeMin = msg->range_min;
+    double minDist = msg->range_max;
+
+    for (int i=0; i<msg->ranges.size(); i++) {
+        angle += increment;
+
+        double r = msg->ranges[i];
```

Comment:
The ranges are 32bit floats, so this should be float instead of double. On the pi this will make a big difference as 64bit math is done in software.

http://docs.ros.org/kinetic/api/sensor_msgs/html/msg/LaserScan.html